### PR TITLE
Gridfeedoc

### PR DESCRIPTION
**Pull request by bedirhankose**
This pull request includes the document I added to the Grid Fee Matrix section of the Simply Simulation documentation. 

1. Added an example table to clearly illustrate the structure of the grid fee matrix.
2. Provided detailed explanations under two main headings for entering the grid fee matrix.
3. Added an example grid fee matrix definition in JSON format.
4. Clearly outlined the method to construct a grid fee matrix from configuration files.
5. Added sample entries for the configuration file.
6. Included a conclusion section at the end of the document.

### CODE REVIEW

File: doc/source/matching_strategies.rst [hunk 1]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
 | Origin Cluster: 1  | 1                        | 0                       |
 +--------------------+--------------------------+-------------------------+
 
+.. _grid_fee_matrix:
+
+Methods to Enter Grid Fee Matrix
+--------------------------------
+
+There are two primary methods to input a grid fee matrix into the Simply Simulation:
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]:
```suggestion
There are two methods to define a grid fee matrix for the simply simulation:
```

File: doc/source/matching_strategies.rst [hunk 2]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
+
+Units
+"""""
+
+- Specify consistent units for both row and column headers to avoid confusion (e.g., USD, EUR).
+
+Weight Factor
+"""""""""""""
+
+- The weight factor describes the relationship between grid fee and cumulative power network edge weights.
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]:
```suggestion
- The weight factor describes the relationship between grid fee and cumulative power network edge weight between two nodes in the network. The cumulative weight is the sum of all edge weights along the shortest path in the network between those nodes. If there are multiple nodes per cluster, nodes of one cluster have equivalent cumulative weights to all nodes of of another cluster.
```

File: doc/source/matching_strategies.rst [hunk 3]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
+         "unit": "$/kWh",
+         "value": 0.05
+       },
+       {
+         "name": "distribution_fee",
+         "unit": "$/kWh",
+         "value": 0.03
+       }
+     ]
+   }
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]: Where can I find this in the code? Is it in another branch?

File: doc/source/matching_strategies.rst [hunk 4]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
+
+Building a Grid Fee Matrix from Config Files
+--------------------------------------------
+
+Composition from Config Files
+""""""""""""""""""""""""""""""
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]: not visible

File: doc/source/matching_strategies.rst [hunk 5]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
+Units
+"""""
+
+- Specify consistent units for both row and column headers to avoid confusion (e.g., USD, EUR).
+
+Weight Factor
+"""""""""""""
+
+- The weight factor describes the relationship between grid fee and cumulative power network edge weights.
+
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]: too many section, does not look very compact, maybe try
- `**Format**: The grid fee matrix should be provided in a structured format such as JSON or CSV.` etc.
- or as a two column table

File: doc/source/matching_strategies.rst [hunk 6]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
+--------------------------------------------
+
+Composition from Config Files
+""""""""""""""""""""""""""""""
+
+The grid fee matrix can be composed by reading entries from configuration files using the following attributes:
+
+- ``default_grid_fee``: Default grid fee used by the market maker.
+- ``local_grid_fee``: Local grid fee to be used.
+- ``weight_factor``: Factor describing the relationship of grid fee to cumulative power network edge weights.
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]:
```suggestion
- ``default_grid_fee``: Default grid fee applied for trades with the market maker
- ``local_grid_fee``: Local grid fee applied for trades within a cluster
- ``weight_factor``: Factor describing the relationship of grid fee to cumulative power network edge weights (see description above)
```

File: doc/source/matching_strategies.rst [hunk 7]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
+   # market
+   #--------------------------
+   # default grid_fee to be used by market maker
+   default_grid_fee = 0.09
+   # local grid fee to be used
+   local_grid_fee = 0
+   # factor describing the relation of grid fee to cumulative power network edge weights
+   weight_factor = 0.03
+
+Entering a grid fee matrix into a simulation is essential for accurate cost calculations related to using the grid. Ensure proper format, identifiers, units, and values for the rate matrix, whether complete or created from config file entries.
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]: This description should stand in the beginning, e.g. after line 52

File: doc/source/matching_strategies.rst [hunk 8]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
+
+- ``default_grid_fee``: Default grid fee used by the market maker.
+- ``local_grid_fee``: Local grid fee to be used.
+- ``weight_factor``: Factor describing the relationship of grid fee to cumulative power network edge weights.
+
+Steps to Build Grid Fee Matrix
+""""""""""""""""""""""""""""""
+
+1. Retrieve ``default_grid_fee``, ``local_grid_fee``, and ``weight_factor`` values from the configuration file.
+2. Use these values to calculate or define the grid fee matrix entries.
```
Comment (j-ti) [inline doc/source/matching_strategies.rst]: This is happening within simply, steps do not need to be explained (i.e. can be removed), ... parameter are already explained above

File: doc/source/matching_strategies.rst [hunk 9]
```diff
@@ -40,6 +40,105 @@ Grid-fees are calculated for each order using the following grid fee matrix:
...
 
+.. _grid_fee_matrix:
+
+Methods to Enter Grid Fee Matrix
+--------------------------------
+
+There are two primary methods to input a grid fee matrix into the Simply Simulation:
+
+1. Providing a Complete Grid Fee Matrix
+2. Building a Grid Fee Matrix from Configuration Files
```
Comment (j-ti) [inline doc/source/matching_strategies.rst:53]: maybe here reference links to the subsections below could be used

**Review by j-ti — CHANGES_REQUESTED** [review]
Some remarks in comments

File: doc/source/matching_strategies.rst (hunk 3 above)
Comment (bedirhankose) [reply doc/source/matching_strategies.rst]: I created this as an example.

**Pull request closed without merging**